src/gamestate/system_state.cpp

```diff
@@ -2915,23 +2915,32 @@ void state::single_game_tick() {
 void state::game_loop() {
 	while(quit_signaled.load(std::memory_order::acquire) == false) {
 		if(network_mode == sys::network_mode::client) {
+			network_state.send_and_receive_commands(*this);
 			command::execute_pending_commands(*this);
 			std::this_thread::sleep_for(std::chrono::milliseconds(15));
 		} else {
 			auto speed = actual_game_speed.load(std::memory_order::acquire);
 			auto upause = ui_pause.load(std::memory_order::acquire);
 			if(speed <= 0 || upause || internally_paused) {
+				network_state.send_and_receive_commands(*this);
 				command::execute_pending_commands(*this);
 				std::this_thread::sleep_for(std::chrono::milliseconds(15));
 			} else {
 				auto entry_time = std::chrono::steady_clock::now();
 				auto ms_count = std::chrono::duration_cast<std::chrono::milliseconds>(entry_time - last_update).count();
+
+				network_state.send_and_receive_commands(*this);
+				command::execute_pending_commands(*this);
 				if(speed >= 5 || ms_count >= game_speed[speed]) { /*enough time has passed*/
-					command::execute_pending_commands(*this);
 					last_update = entry_time;
-					single_game_tick();
+					if(network_mode == sys::network_mode::host) {
+						command::payload c;
+						c.type = command::command_type::advance_tick;
+						outgoing_commands.push(c);
+					} else {
+						single_game_tick();
+					}
 				} else {
-					command::execute_pending_commands(*this);
```

Comment:
why has this gone away?

Reply:
it went above

Reply:
but above, it will only be executed if the tick is ready to advance, meaning that mid tick commands would have to wait, no?

Reply:
no, it will execute regardless of whenever it's mid tick or when it's supposed to go in a tick